src/main/java/com/google/sps/storage/BookmarkStorageManager.java

```diff
@@ -0,0 +1,112 @@
+        Filter userFilter = new FilterPredicate(BOOKMARK_EMAIL_PROPERTY, FilterOperator.EQUAL, email);
+        Filter videoIdFilter = new FilterPredicate(BOOKMARK_VIDEOID_PROPERTY, FilterOperator.EQUAL, videoId);
+        Filter bookmarkFilter = CompositeFilterOperator.and(userFilter, videoIdFilter);
+        Query query = new Query(ENTITY_BOOKMARK).setFilter(bookmarkFilter);
+        
+        try {
+            PreparedQuery results = datastore.prepare(query);
+            List<Bookmark> bookmarks = new ArrayList<>();
+
+            // Creates a list of Bookmarks from the perpared query results
+            for (Entity entity : results.asIterable()) {
+                String id = KeyFactory.keyToString(entity.getKey());
+                long timestamp = (long) entity.getProperty(BOOKMARK_TIMESTAMP_PROPERTY);
+                String title = (String) entity.getProperty(BOOKMARK_TITLE_PROPERTY);
+                String content = (String) entity.getProperty(BOOKMARK_CONTENT_PROPERTY);
+
+                Bookmark bookmark = new Bookmark(id, timestamp, title, content);
+                bookmarks.add(bookmark);
+            }
+            return bookmarks;
+        } catch (Exception e) {
+            throw new BookmarkStorageException(EXCEPTION_GET_BOOKMARKS, e.getMessage(), e.getCause());
+        }
+
+    }
+
+    /**
+     * Adds a bookmark with the defined parameters to the Datastore database
+     * @param userEmail The user email for the bookmark
+     * @param videoId The Youtube video id for the bookmark
+     * @param videoTimestamp The Youtube timestamp for the bookmark
+     * @param title The title of the bookmark
+     * @param content The content of the bookmark
+     */
+    public void addBookmark(String userEmail, String videoId, long videoTimestamp, String title, String content) throws BookmarkStorageException {
+        // Tries to create an Entity object and store it in Datastore
+        Entity bookmark;
+        try {
+            bookmark = new Entity(ENTITY_BOOKMARK);
+            bookmark.setProperty(BOOKMARK_EMAIL_PROPERTY, userEmail);
+            bookmark.setProperty(BOOKMARK_VIDEOID_PROPERTY, videoId);
+            bookmark.setProperty(BOOKMARK_TIMESTAMP_PROPERTY, videoTimestamp);
+            bookmark.setProperty(BOOKMARK_TITLE_PROPERTY, title);
+            bookmark.setProperty(BOOKMARK_CONTENT_PROPERTY, content);
+
+            datastore.put(bookmark);
+        } catch (Exception e) {
+            throw new BookmarkStorageException(EXCEPTION_ADD_BOOKMARK, e.getMessage(), e.getCause());
+        }
+    }
+
+    /**
+     * Removes a bookmark from the database
+     * @param id The globally unique id of the bookmark to remove
+     */
+    public void removeBookmark(String id) throws BookmarkStorageException {
+        // Tries to parse the Entity Key from the string to remove the Entity from Datastore
+        Key key;
+        try {
+            key = KeyFactory.stringToKey(id);
```

Comment:
nit: i think you can just do `Key key = KeyFactory.stringToKey(id)` instead (just saving the one `Key key;` line)